pymeasure/instruments/proterial/rod4.py

```diff
@@ -0,0 +1,144 @@
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+import logging
+
+from pymeasure.instruments import Instrument, Channel
+from pymeasure.instruments.validators import (truncated_range,
+                                              strict_discrete_set)
+
+
+log = logging.getLogger(__name__)
+log.addHandler(logging.NullHandler())
+
+
+class ROD4Channel(Channel):
+    """Implementation of a ROD-4 MFC channel."""
+
+    actual_flow = Channel.measurement(
+        "\x020{ch}RFX",
+        """Measure the actual flow in % ."""
+    )
+    setpoint = Channel.control(
+        "\x020{ch}RFD", "\x020{ch}SFD%.1f",
+        """Control the set point in % of control range.""",
+        validator=truncated_range,
+        values=[0, 100],
+    )
+    mfc_range = Channel.control(
+        "\x020{ch}RFK", "\x020{ch}SFK%d",
+        """Control the MFC range in sccm.
+        Upper limit is 200 slm.""",
+        validator=truncated_range,
+        values=[0, 200000]
+    )
+    ramp_time = Channel.control(
+        "\x020{ch}RRT", "\x020{ch}SRT%.1f",
+        """Control the MFC set point ramping time in seconds.""",
+        validator=truncated_range,
+        values=[0, 200000]
+    )
+    valve_mode = Channel.control(
+        "\x020{ch}RVM", "\x020{ch}SVM%d",
+        """Control the MFC valve mode.
+        Valid options are `flow`, `close`, and `open`. """,
+        validator=strict_discrete_set,
+        values={'flow': 0, 'close': 1, 'open': 2},
+        map_values=True
+    )
+
+    def __init__(self, *argv, **kwargs):
+        super().__init__(*argv, **kwargs)
+        self._flow_unit = None
+
+    @property
+    def flow_unit(self):
+        """Control flow units for the selected channel.
+        Valid options are %, sccm, or slm.
+        Display in absolute units is in sccm for control range < 10 slm."""
+        if self._flow_unit is None:
+            log.info("Flow units have not been set on ROD-4 channel "
+                     f"{self.id}")
+        return self._flow_unit
+
+    @flow_unit.setter
+    def flow_unit(self, units):
+        values = {'%': 0, 'sccm': 1, 'slm': 1}
+        if isinstance(units, str) and units.casefold() in values:
+            units = units.casefold()
+            self.write("\x020{ch}SFU%d" % values[units])
```

Comment:
You could use normal f-strings:
```suggestion
            self.write(f"\x020{{ch}}SFU{value[units]}")
```
If you use double curly braces, they will be reduced by one, but not evaluated.

Reply:
Changed this to a "setting" property, removed the need for f-string